src/ILCompiler.Compiler/src/Compiler/DependencyAnalysis/NodeFactory.cs

```diff
@@ -254,10 +254,33 @@ public ISymbolNode MethodEntrypoint(MethodDesc method)
             return _methodCode.GetOrAdd(method);
         }
 
-        public ISymbolNode WellKnownEntrypoint(WellKnownEntrypoint entrypoint)
+        static readonly string[][] s_helperEntrypointNames = new string[][] {
+            new string[] { "System.Runtime.CompilerServices", "CctorHelper", "CheckStaticClassConstructionReturnGCStaticBase" },
+            new string[] { "System.Runtime.CompilerServices", "CctorHelper", "CheckStaticClassConstructionReturnNonGCStaticBase" }
+        };
+
+        static ISymbolNode[] s_helperEntrypointSymbols;
```

Comment:
This should not be static - copy&paste from s_helperEntrypointNames. Are you planning to fix it?